src/macos/simulate.rs

```diff
@@ -1,112 +1,92 @@
 use crate::rdev::{Button, EventType, SimulateError};
 use core_graphics::event::{
-    CGEvent, CGEventTapLocation, CGEventType, CGKeyCode, CGMouseButton, ScrollEventUnit,
+    CGEvent, CGEventTapLocation, CGEventType, CGMouseButton, ScrollEventUnit,
 };
 use core_graphics::event_source::{CGEventSource, CGEventSourceStateID};
 use core_graphics::geometry::CGPoint;
+use std::convert::TryInto;
 
 use crate::macos::keycodes::code_from_key;
 
 unsafe fn convert_native_with_source(
     event_type: &EventType,
     source: CGEventSource,
-) -> Result<CGEvent, ()> {
+) -> Option<CGEvent> {
     match event_type {
         EventType::KeyPress(key) => {
-            if let Some(code) = code_from_key(*key) {
-                return CGEvent::new_keyboard_event(source, code as CGKeyCode, true);
-            }
-            Err(())
+            CGEvent::new_keyboard_event(source, code_from_key(*key)?, true).ok()
```

Comment:
I'm not a big fan of clever oneliners.

Can we keep
 ```rust
let code = code_from_key(*key)?;
CGEvent::new_keyboard_event(source, code, true).ok()
```
? Seems more readable.